discovery_engine_core/ai/xayn-ai/src/ranker/system.rs

```diff
@@ -47,7 +52,7 @@ pub(super) struct State {
     pub(super) user_interests: UserInterests,
 
     /// Key phrases.
-    pub(super) relevances: RelevanceMap,
+    pub(super) key_phrases: KeyPhrases,
```

Comment:
:wrench: we need to change `serialize`/`deserialize`.

Reply:
i don't understand what you are trying to say. the new `KeyPhrases` are implementing ser/de. is the way how `KeyPhrases` implements ser/de not the correct one?

Reply:
We are changing the state so we need to bump the version number for the de/serialization and migrate the content during deserialization

Reply:
i see. afaik we only have beta user currently, so do we actually need a full scale migration scheme?

i'll bump the version and add a outdated version handler for now. otherwise we would need to follow deep down the rabit hole and introduce versioning like `obake` for the `State` including an actual migration scheme from `RelevanceMap` to `KeyPhrases`.

Reply:
for outdated state versions i changed the error to the default state instead.